internal/controller/watcher_controller.go

```diff
@@ -151,7 +152,7 @@ func (r *WatcherReconciler) handleProcessingState(ctx context.Context,
 		return r.updateWatcherState(ctx, watcherCR, shared.StateError, err)
 	}
 
-	virtualSvc, err := istio.NewVirtualService(r.WatcherVSNamespace, watcherCR, gateways)
+	virtualSvc, err := r.VirtualServiceFactory.NewVirtualService(r.WatcherVSNamespace, watcherCR, gateways)
```

Comment:
Watcher and VirtualService will not be applied to the same namespace or?

Reply:
There is a dedicated FlagVar [`flagVar.IstioGatewayNamespace`](https://github.com/kyma-project/lifecycle-manager/blob/main/cmd/main.go#L392) used. Right now it is configured to default to [`kcp-system`](https://github.com/kyma-project/lifecycle-manager/blob/main/internal/pkg/flags/flags.go#L37). So Watcher and VirtualService reside in the same namespace as of now, but not sure what the origin reason for the flagvar is and if this may need to change in the future.

Reply:
Just also remembered: https://github.com/kyma-project/lifecycle-manager/pull/1349#discussion_r1506042184

Actually, we have an implicit constraint now that `r.WatcherVSNamespace == watcher.Namespace`. So we may even get rid of this argument. Maybe also the flagvar needs to be renamed entirely

Reply:
That's why I am asking yes. 😄 

Seems the flag is then obsolet even and can be removed. And Watcher namespace always dictates the VS namespace, as you said.

Reply:
Removed explicit namespace arg with https://github.com/kyma-project/lifecycle-manager/pull/1349/commits/443c8b87fcf54c486833401ae256729153110e38

Overall `flagVar.IstioGatewayNamespace` to be discussed in the daily.